kof-operator/internal/controller/clusterdeployment_controller_test.go

```diff
@@ -71,40 +92,81 @@ var _ = Describe("ClusterDeployment Controller", func() {
 			if err != nil && errors.IsNotFound(err) {
 				Expect(k8sClient.Create(ctx, certNamespace)).To(Succeed())
 			}
+
 			By("creating the custom resource for the Kind ClusterDeployment")
 			err = k8sClient.Get(ctx, clusterDeploymentNamespacedName, clusterDeployment)
 			if err != nil && errors.IsNotFound(err) {
 				resource := &kcmv1alpha1.ClusterDeployment{
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      clusterDeploymentName,
-						Namespace: "default",
-						Labels:    make(map[string]string),
+						Namespace: DEFAULT_NAMESPACE,
+						Labels:    map[string]string{},
 					},
 					Spec: kcmv1alpha1.ClusterDeploymentSpec{
 						Template: "test-cluster-template",
 						Config:   &apiextensionsv1.JSON{Raw: []byte(clusterLabels)},
 					},
 				}
 				Expect(k8sClient.Create(ctx, resource)).To(Succeed())
+
+				resource.Status = kcmv1alpha1.ClusterDeploymentStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:               kcmv1alpha1.ReadyCondition,
+							Status:             metav1.ConditionTrue,
+							LastTransitionTime: metav1.Time{Time: time.Now()},
+							Reason:             "ClusterReady",
+							Message:            "Cluster is ready",
+						},
+					},
+				}
+				k8sClient.Status().Update(ctx, resource)
+			}
+
+			By("creating the fake Secret for the cluster deployment kubeconfig")
+			err = k8sClient.Get(ctx, kubeconfigSecretNamespacesName, kubeconfigSecret)
+			if err != nil && errors.IsNotFound(err) {
+				resource := &coreV1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secretName,
+						Namespace: DEFAULT_NAMESPACE,
+						Labels:    map[string]string{},
+					},
+					Data: map[string][]byte{
+						"value": []byte("aGVsbG8gd29ybGQ="),
```

Comment:
No reason to put any value, then better to keep it's empty
```suggestion
						"value": []byte(""),
```